# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Lok997/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My lane: Search performance and content quality

I want to investigate which observable search-page factors are associated with stronger click-through performance. This matters because understanding these patterns can help prioritize which pages or content types may deserve attention. I want to use the available data to identify useful patterns without claiming that the analysis proves a causal relationship.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research question: Among pages with meaningful search visibility, which observable factors are associated with higher click-through rate (CTR), and how can these patterns help prioritize pages for further review?

The decision is which pages or content types should be prioritized for investigation and improvement. A search or content team would act on the recommendation. A wrong recommendation could waste time on low-impact pages, overlook pages with genuine improvement opportunities, or lead to changes that do not improve performance. The analysis is directional and does not prove causation.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

url = "https://raw.githubusercontent.com/Lok997/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of pages:", len(df))
print("Median CTR:", df["ctr"].median())
print("Median average position:", df["avg_position"].median())
print("Median impressions over 90 days:", df["impressions_90d"].median())


Number of pages: 30000
Median CTR: 0.07
Median average position: 10.8
Median impressions over 90 days: 731.0


The starter dataset contains 30,000 pages. The median CTR is 7%, the median average position is 10.8, and the median number of impressions over 90 days is 731. These numbers show that there is enough variation in search visibility and click performance to investigate which observable factors are associated with stronger CTR.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis can identify observed patterns and directional relationships in the available data, and it can support decisions about which factors may be worth investigating further. It cannot prove that one factor causes another, and it cannot guarantee future CTR or predict Google's ranking decisions. The findings should therefore be treated as decision support, not causal proof.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Claim boundary check:")
print("Can claim: observed patterns and directional relationships.")
print("Cannot claim: causality, guaranteed future CTR, or Google's ranking decisions.")


Claim boundary check:
Can claim: observed patterns and directional relationships.
Cannot claim: causality, guaranteed future CTR, or Google's ranking decisions.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.